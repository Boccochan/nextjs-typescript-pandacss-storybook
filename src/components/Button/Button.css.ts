import { cva } from "#/styled-system/css";

export const button = cva({
  variants: {
    color: {
      primary: {
        bg: "btn.primary.bg",
        color: "btn.primary.text",
        _osDark: {
          bg: "btn.primary.bg.dark",
          color: "btn.primary.text.dark",
        },
      },
      secondary: {
        bg: "btn.danger.bg",
        color: "btn.danger.text",
        _osDark: {
          bg: "btn.danger.bg.dark",
          color: "btn.danger.text.dark",
        },
      },
    },
  },
});
