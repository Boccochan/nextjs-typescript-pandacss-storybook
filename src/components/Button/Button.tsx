"use client";
import { css } from "../../../styled-system/css";

export const Button = () => {
  return <button className={css({ bg: "btn.primary" })}>hello world</button>;
};
