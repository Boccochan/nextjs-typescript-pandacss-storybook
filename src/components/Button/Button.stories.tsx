import { action } from "@storybook/addon-actions";
import type { Meta, StoryObj } from "@storybook/react";
import { useTranslations } from "next-intl";
import type { ComponentProps } from "react";

import { Button } from "./Button";
type ButtonProps = ComponentProps<typeof Button>;

const ButtonWithIntl = ({ color, size, label, ...rest }: ButtonProps) => {
  const t = useTranslations();
  return <Button color={color} size={size} label={t(label)} {...rest} />;
};

const meta = {
  title: "components/Button",
  component: ButtonWithIntl,
  parameters: {
    layout: "centered",
  },
  argTypes: {
    color: { control: "select", options: ["primary", "danger"] },
    size: { control: "select", options: ["md", "lg"] },
  },
} satisfies Meta<typeof ButtonWithIntl>;

export default meta;
type Story = StoryObj<typeof ButtonWithIntl>;

export const Primary: Story = {
  args: {
    color: "primary",
    size: "md",
    label: "hello",
    disabled: false,
    loading: false,
    onSubmit: action("clicked"),
  },
};

export const Danger: Story = {
  args: {
    color: "danger",
    size: "md",
    label: "hello",
    disabled: false,
    loading: false,
    onSubmit: action("clicked"),
  },
};
