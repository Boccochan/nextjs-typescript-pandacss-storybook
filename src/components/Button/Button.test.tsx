import { render } from "@testing-library/react";
import React from "react";

import { Button } from "./Button";

describe("Snapshot button", () => {
  it("Primary", async () => {
    const button = render(<Button label="hello" />);

    expect(button).toMatchSnapshot();
  });

  it("Danger", async () => {
    const button = render(<Button label="hello" color="danger" />);

    expect(button).toMatchSnapshot();
  });

  it("Size=sm", async () => {
    const button = render(<Button label="hello" size="sm" />);

    expect(button).toMatchSnapshot();
  });

  it("Size=md", async () => {
    const button = render(<Button label="hello" size="md" />);

    expect(button).toMatchSnapshot();
  });

  it("Size=lg", async () => {
    const button = render(<Button label="hello" size="lg" />);

    expect(button).toMatchSnapshot();
  });

  it("Size=xl", async () => {
    const button = render(<Button label="hello" size="xl" />);

    expect(button).toMatchSnapshot();
  });

  it("Loading", async () => {
    const button = render(<Button label="hello" loading={true} />);

    expect(button).toMatchSnapshot();
  });

  it("Disabled", async () => {
    const button = render(<Button label="hello" disabled={true} />);

    expect(button).toMatchSnapshot();
  });
});
