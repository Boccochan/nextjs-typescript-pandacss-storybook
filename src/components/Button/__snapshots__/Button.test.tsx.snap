// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Button Danger 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_danger hover:bg_danger.hover text_light px_4 rounded_md fs_md min-w_4rem h_2.4rem"
      >
        <div
          class="d_flex justify_center items_center gap_2"
        >
          <span>
            hello
          </span>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_danger hover:bg_danger.hover text_light px_4 rounded_md fs_md min-w_4rem h_2.4rem"
    >
      <div
        class="d_flex justify_center items_center gap_2"
      >
        <span>
          hello
        </span>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot Button Disabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-disabled="true"
        class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_4 rounded_md fs_md min-w_4rem h_2.4rem"
        disabled=""
      >
        <div
          class="d_flex justify_center items_center gap_2"
        >
          <span>
            hello
          </span>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-disabled="true"
      class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_4 rounded_md fs_md min-w_4rem h_2.4rem"
      disabled=""
    >
      <div
        class="d_flex justify_center items_center gap_2"
      >
        <span>
          hello
        </span>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot Button Loading 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-disabled="true"
        class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_4 rounded_md fs_md min-w_4rem h_2.4rem"
        disabled=""
      >
        <div
          class="pos_absolute inset_0 m_auto w_fit-content h_full d_flex justify_center items_center"
        >
          <div
            aria-label="loading"
            class="fs_6.8px pos_relative transform_translateZ(0) animation_spinner_1.1s_infinite_linear rounded_50% w_3em h_3em border-t_0.3em_solid_{colors.light} border-r_0.3em_solid_{colors.light} border-b_0.3em_solid_{colors.light} border-l_0.3em_solid_rgba(255,_255,_255,_0)"
            role="alert"
          />
        </div>
        <div
          class="d_flex justify_center items_center gap_2 vis_hidden"
        >
          <span>
            hello
          </span>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-disabled="true"
      class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_4 rounded_md fs_md min-w_4rem h_2.4rem"
      disabled=""
    >
      <div
        class="pos_absolute inset_0 m_auto w_fit-content h_full d_flex justify_center items_center"
      >
        <div
          aria-label="loading"
          class="fs_6.8px pos_relative transform_translateZ(0) animation_spinner_1.1s_infinite_linear rounded_50% w_3em h_3em border-t_0.3em_solid_{colors.light} border-r_0.3em_solid_{colors.light} border-b_0.3em_solid_{colors.light} border-l_0.3em_solid_rgba(255,_255,_255,_0)"
          role="alert"
        />
      </div>
      <div
        class="d_flex justify_center items_center gap_2 vis_hidden"
      >
        <span>
          hello
        </span>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot Button Primary 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_4 rounded_md fs_md min-w_4rem h_2.4rem"
      >
        <div
          class="d_flex justify_center items_center gap_2"
        >
          <span>
            hello
          </span>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_4 rounded_md fs_md min-w_4rem h_2.4rem"
    >
      <div
        class="d_flex justify_center items_center gap_2"
      >
        <span>
          hello
        </span>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot Button Size=lg 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_5 rounded_lg fs_lg min-w_5rem h_2.8rem"
      >
        <div
          class="d_flex justify_center items_center gap_2"
        >
          <span>
            hello
          </span>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_5 rounded_lg fs_lg min-w_5rem h_2.8rem"
    >
      <div
        class="d_flex justify_center items_center gap_2"
      >
        <span>
          hello
        </span>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot Button Size=md 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_4 rounded_md fs_md min-w_4rem h_2.4rem"
      >
        <div
          class="d_flex justify_center items_center gap_2"
        >
          <span>
            hello
          </span>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_4 rounded_md fs_md min-w_4rem h_2.4rem"
    >
      <div
        class="d_flex justify_center items_center gap_2"
      >
        <span>
          hello
        </span>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot Button Size=sm 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_3 rounded_sm fs_sm min-w_3rem h_2rem"
      >
        <div
          class="d_flex justify_center items_center gap_2"
        >
          <span>
            hello
          </span>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_3 rounded_sm fs_sm min-w_3rem h_2rem"
    >
      <div
        class="d_flex justify_center items_center gap_2"
      >
        <span>
          hello
        </span>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot Button Size=xl 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_6 rounded_xl fs_xl min-w_6rem h_3.2rem"
      >
        <div
          class="d_flex justify_center items_center gap_2"
        >
          <span>
            hello
          </span>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_6 rounded_xl fs_xl min-w_6rem h_3.2rem"
    >
      <div
        class="d_flex justify_center items_center gap_2"
      >
        <span>
          hello
        </span>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot Button Size=xs 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_2 rounded_xs fs_xs min-w_2rem h_1.6rem"
      >
        <div
          class="d_flex justify_center items_center gap_2"
        >
          <span>
            hello
          </span>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_2 rounded_xs fs_xs min-w_2rem h_1.6rem"
    >
      <div
        class="d_flex justify_center items_center gap_2"
      >
        <span>
          hello
        </span>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot Button With Md icon 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_4 rounded_md fs_md min-w_4rem h_2.4rem"
      >
        <div
          class="d_flex justify_center items_center gap_2"
        >
          <svg
            aria-hidden="true"
            class="mt_0.1rem"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          <span>
            hello
          </span>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="cursor_pointer white-space_nowrap font_medium disabled:opacity_0.4 disabled:pointer-events_none active:opacity_0.9 pos_relative bg_primary hover:bg_primary.hover text_light px_4 rounded_md fs_md min-w_4rem h_2.4rem"
    >
      <div
        class="d_flex justify_center items_center gap_2"
      >
        <svg
          aria-hidden="true"
          class="mt_0.1rem"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
        <span>
          hello
        </span>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
